paf-mvp-core-js/src/express/node.ts

```diff
@@ -142,74 +169,72 @@ export class Node implements INode {
    * against a JSON schema.
    * @returns the built handler
    */
-  checkJsonBody =
-    (jsonSchema: JsonSchemaType): RequestHandler =>
-    (req: Request, res: Response, next: NextFunction) => {
-      const validation = this.jsonValidator.validate(jsonSchema, req.body as string);
-      if (!validation.isValid) {
-        const details = validation.errors.map((e) => e.message).join(' - ');
-        const error: NodeError = {
-          type: NodeErrorType.INVALID_JSON_BODY,
-          details,
-        };
-        res.status(400);
-        res.json(error);
-        next(error);
-      } else {
-        next();
-      }
-    };
+  checkJsonBody = (req: Request, res: Response, next: NextFunction) => {
+    const { jsonSchemaName } = this.getContext(res);
+    const validation = this.jsonValidator.validate(jsonSchemaName, req.body as string);
+    if (!validation.isValid) {
+      const details = validation.errors.map((e) => e.message).join(' - ');
+      const error: NodeError = {
+        type: NodeErrorType.INVALID_JSON_BODY,
+        details,
+      };
+      res.status(400);
+      res.json(error);
+      next(error);
+    } else {
+      next();
+    }
+  };
 
   /**
    * Builds a handler that validates the query string
    * against a JSON schema.
    * @returns the built handler
    */
-  checkQueryString =
-    (jsonSchema: JsonSchemaType, isRedirect: boolean): RequestHandler =>
-    (req: Request, res: Response, next: NextFunction) => {
-      const data = req.query[QSParam.paf] as string | undefined;
-      const decodedData = data ? decodeBase64(data) : undefined;
-      if (!decodedData) {
+  checkQueryString = (req: Request, res: Response, next: NextFunction) => {
+    const { isRedirect, jsonSchemaName } = this.getContext(res);
+    const data = req.query[QSParam.paf] as string | undefined;
+    const decodedData = data ? decodeBase64(data) : undefined;
+    if (!decodedData) {
+      const error: NodeError = {
+        type: NodeErrorType.INVALID_QUERY_STRING,
```

Comment:
Don't be fooled by GitHub diff: the only change here is line 169, where we remove the `(jsonSchema: JsonSchemaType, isRedirect: boolean)` parameters